pylsp/workspace.py

```diff
@@ -144,30 +157,21 @@ def report_progress(
             client_supports_progress_reporting = False
 
         if client_supports_progress_reporting:
-            try:
-                token = self._progress_begin(title, message, percentage)
-            except Exception:  # pylint: disable=broad-exception-caught
-                log.warning(
-                    "There was an error while trying to initialize progress reporting."
-                    "Likely progress reporting was used in a synchronous LSP handler, "
-                    "which is not supported by progress reporting yet.",
-                    exc_info=True
-                )
+            token = self._progress_begin(title, message, percentage, skip_token_initialization)
 
-            else:
-                def progress_message(message: str, percentage: Optional[int] = None) -> None:
-                    self._progress_report(token, message, percentage)
+            def progress_message(message: str, percentage: Optional[int] = None) -> None:
+                self._progress_report(token, message, percentage)
 
-                try:
-                    yield progress_message
-                finally:
-                    self._progress_end(token)
+            try:
+                yield progress_message
+            finally:
+                self._progress_end(token)
 
-                return
+            return
 
         # FALLBACK:
-        # If the client doesn't support progress reporting, or if we failed to
-        # initialize it, we have a dummy method for the caller to use.
+        # If the client doesn't support progress reporting,
+        # we have a dummy method for the caller to use.
```

Comment:
```suggestion
        # If the client doesn't support progress reporting, we have a dummy method
        # for the caller to use.
```

Formatting improvement.